internal/display/bufwindow.go

```diff
@@ -601,7 +601,7 @@ func (w *BufWindow) displayBuffer() {
 		}
 
 		var word []glyph
-		if wordwrap {
+		if wordwrap && w.bufWidth > 0 {
```

Comment:
I'm guilty of this bug, I admit it.

But it seems the more correct fix is:

```
--- a/internal/display/bufwindow.go
+++ b/internal/display/bufwindow.go
@@ -152,6 +152,9 @@ func (w *BufWindow) updateDisplayInfo() {
 	if w.Buf.Settings["scrollbar"].(bool) && w.Buf.LinesNum() > w.Height {
 		w.bufWidth--
 	}
+	if w.bufWidth < 0 {
+		w.bufWidth = 0
+	}
 
 	if w.bufWidth != prevBufWidth && w.Buf.Settings["softwrap"].(bool) {
 		for _, c := range w.Buf.GetCursors() {
```

Reply:
I mean, the crash actually happens not when `bufWidth` becomes zero but when it becomes negative. (Slice cap can be zero, but it cannot be negative.)

And BTW, the root cause of it is that `gutterOffset` is bigger than the window width, which also causes the ruler overwriting the other split pane (as you probably noticed). I'm now working on the fix for that, which probably will also fix this crash.

Reply:
Here it is: #3069

Reply:
> [...] as you probably noticed [...]

Yes I did. Any way, we go with yours.